rewrite-gradle/src/test/java/org/openrewrite/gradle/trait/GradleDependencyTest.java

```diff
@@ -69,9 +71,102 @@ void literal() {
         );
     }
 
+    @ParameterizedTest
+    @ValueSource(strings = {
+      "api",
+      "implementation",
+      "compileOnly",
+      "runtimeOnly",
+      "testImplementation",
+      "testCompileOnly",
+      "testRuntimeOnly",
+      "debugImplementation",
+      "releaseImplementation",
+      "androidTestImplementation",
+      "featureImplementation",
+      "annotationProcessor",
+      "kapt",
+      "ksp"
+    })
+    void methods(String method) {
+        rewriteRun(
+          spec -> spec.beforeRecipe(withToolingApi()),
+          buildGradle(
+            """
+              plugins {
+                  id "java"
+              }
+              
+              repositories {
+                  mavenCentral()
+              }
+              
+              dependencies {
+                  %s "com.google.guava:guava:28.2-jre"
+              }
+              """.formatted(method),
+            """
+              plugins {
+                  id "java"
+              }
+              
+              repositories {
+                  mavenCentral()
+              }
+              
+              dependencies {
+                  /*~~(com.google.guava:guava:28.2-jre)~~>*/%s "com.google.guava:guava:28.2-jre"
+              }
+              """.formatted(method)
+          )
+        );
+    }
+
+    @ParameterizedTest
+    @ValueSource(strings = {
+      "compile", // deprecated
+      "runtime", // deprecated
+      "testCompile", // deprecated
+      "testRuntime" // deprecated
+    })
+    void decprecatedMethods(String method) {
```

Comment:
This is presently failing because Gradle 6.9.4 supports Java 15 at the time of its release... We may need to think about how to deal with these sorts of things...